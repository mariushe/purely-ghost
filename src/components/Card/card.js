import React from 'react';
import extractImage from '../../utils/extractImage';


const Card = ({post}) => {


    const image = {
        backgroundImage: `url(${extractImage(post.content)})`,
        backgroundSize: '100%',
        height: '150px',
        borderTopRightRadius: '4px',
        borderTopLeftRadius: '4px',
    };

    const author = {
        fontSize: '15px',
        marginLeft: '10px',
        marginRight: '10px',
        marginTop: '20px',
        color: '#2b313e',
        opacity: '0.9'
    };

    const title = {
        fontSize: '21px',
        marginLeft: '10px',
        marginRight: '10px',
        marginTop: '10px',
        marginBottom: '20px',
        color: '#2b313e'
    };

    const card = {
        borderRadius: '4px',
        boxShadow: '0 1px 7px rgba(0,0,0,.05)',
        border: '1px solid rgba(0,0,0,.04)!important',
        background: '#fff',
        flexBasis: '300px',
        marginBottom: '20px',

    };

    return (
        <div style={card}>
            <a href={post.link}>
                <div style={image}>
                </div>
                <div style={author}>
                    by {post.creator}
                </div>
                <div style={title}>
                    {post.title}
                </div>
            </a>
        </div>
    );
};

export default Card