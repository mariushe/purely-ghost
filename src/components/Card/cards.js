import React from 'react';
import Card from './Card';


const Cards = ({posts, nrOfPosts}) => {

    const style = {
        display: 'flex',
        flexFlow: 'row wrap',
        justifyContent: 'space-around',
    };

    const CardsToDisplay = posts.slice(0, nrOfPosts).map(post => {
        return <Card key={post.guid} post={post}/>;
    });

    return (
        <div style={style}>
            {CardsToDisplay}
        </div>
    );
};

export default Cards