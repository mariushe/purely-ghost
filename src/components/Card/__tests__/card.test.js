import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../card';


test('Test that Card renders correctly', () => {

    const post = {
        link: "http://www.someurl.com",
        creator: "The Author",
        title: "The title",
        content: "<h1>The Content</h1><img src=\"www.someaddr.com\" alt=\"alt text\"><p>more content</p>"
    };

    const component = renderer.create(
        <Card post={post}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});