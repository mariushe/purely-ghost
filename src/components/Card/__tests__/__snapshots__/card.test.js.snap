// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that Card renders correctly 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "border": "1px solid rgba(0,0,0,.04)!important",
      "borderRadius": "4px",
      "boxShadow": "0 1px 7px rgba(0,0,0,.05)",
      "flexBasis": "300px",
      "marginBottom": "20px",
    }
  }
>
  <a
    href="http://www.someurl.com"
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundImage": "url(www.someaddr.com)",
          "backgroundSize": "100%",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "height": "150px",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#2b313e",
          "fontSize": "15px",
          "marginLeft": "10px",
          "marginRight": "10px",
          "marginTop": "20px",
          "opacity": "0.9",
        }
      }
    >
      by 
      The Author
    </div>
    <div
      style={
        Object {
          "color": "#2b313e",
          "fontSize": "21px",
          "marginBottom": "20px",
          "marginLeft": "10px",
          "marginRight": "10px",
          "marginTop": "10px",
        }
      }
    >
      The title
    </div>
  </a>
</div>
`;
